interfaces/builtin/audio_playback.go

```diff
@@ -159,6 +184,10 @@ func (iface *audioPlaybackInterface) UDevPermanentSlot(spec *udev.Specification,
 
 func (iface *audioPlaybackInterface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
 	spec.AddSnippet(audioPlaybackPermanentSlotAppArmor)
+	if !implicitSystemPermanentSlot(slot) {
+		// This allows to share screen in Core Desktop
+		spec.AddSnippet(audioPlaybackPermanentSlotAppArmorCore)
```

Comment:
Are these rules really Core only? They seem like they'd be relevant to any application snap providing an `audio-playback` slot on core or classic systems. And the `!implicitSystemPermanentSlot(slot)` check isn't going to distinguish between those cases.

Maybe you could just add the `audioPlaybackPermanentSlotAppArmorCore` rules to `audioPlaybackPermanentSlotAppArmor`?

Reply:
You mean to just remove the `!implicitSystemPermanentSlot(slot)` check?

Reply:
And add the `audioPlaybackPermanentSlotAppArmorCore` rules to the end of `audioPlaybackPermanentSlotAppArmor`. No need for two constants if we're always adding the rules together.

Reply:
You could wrap the whole lot in `if !implcitSystemPermanentSlot(slot)`, but I don't think the `*PermanentSlot()` methods ever get used for the implicit slots.

Reply:
(confirmed: it works)

Reply:
> You could wrap the whole lot in `if !implcitSystemPermanentSlot(slot)`, but I don't think the `*PermanentSlot()` methods ever get used for the implicit slots.

I tried adding a `fmt.Println()` in the `PermanentSlot()` method, launched snapd in my Ubuntu system, and it printed it once, for the snapd snap... So it seems that we can remove that.

Reply:
Done, and also added tests for pulseaudio.